src/main/java/io/vertx/core/net/impl/KeyStoreHelper.java

```diff
@@ -112,34 +118,118 @@ public static KeyStoreHelper create(VertxInternal vertx, TrustOptions options) {
           map(path -> vertx.resolveFile(path).getAbsolutePath()).
           map(vertx.fileSystem()::readFileBlocking);
       certValues = Stream.concat(certValues, trustOptions.getCertValues().stream());
-      return new CA(certValues);
+      return new KeyStoreHelper(loadCA(certValues), null);
     } else {
       return null;
     }
   }
 
-  protected final String password;
+  private final String password;
+  private final KeyStore store;
+  private final Map<String, X509KeyManager> wildcardMgrMap = new HashMap<>();
+  private final Map<String, X509KeyManager> mgrMap = new HashMap<>();
 
-  public KeyStoreHelper(String password) {
+  public KeyStoreHelper(KeyStore ks, String password) throws Exception {
+    Enumeration<String> en = ks.aliases();
+    while (en.hasMoreElements()) {
+      String alias = en.nextElement();
```

Comment:
I'm not very good at KeyStores yet; is it possible for multiple aliases to refer to the same certificate? In that case I guess we could optimize by skipping certificates for which we have already stored in `mgrMap` and `wildcardMgrMap`?